lib/src/components/NxDrawer/__tests__/NxDrawer.test.tsx

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { render, fireEvent, within, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
+
+import NxDrawer, { Props } from '../NxDrawer';
+import NxButton from '../../NxButton/NxButton';
+import useToggle from '../../../util/useToggle';
+
+describe('NxDrawer', function() {
+  const minimalProps: Props = {
+    open: true,
+    onClose: () => {},
+    children: 'Drawer Content'
+  };
+
+  const quickRender = rtlRender<React.ComponentPropsWithRef<typeof NxDrawer>>(NxDrawer, minimalProps);
+  const getDrawer = rtlRenderElement(NxDrawer, minimalProps);
+
+  it('renders <dialog> with class nx-drawer', function () {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const dialog = getDrawer()!;
+    expect(dialog.nodeName).toBe('DIALOG');
+    expect(dialog).toHaveClass('nx-drawer');
+  });
+
+  it('has a dialog element with aria-modal set to false', function() {
+    const dialog = getDrawer();
+    expect(dialog).toHaveAttribute('aria-modal', 'false');
+  });
+
+  it('it does not have nx-drawer--closing class when initially loaded', function () {
```

Comment:
Hmm I wonder how we can really test the animations. The classes are implementation details, as can be seen by how much they've repeatedly changed between my commits and yours. I'm not inclined to test the classes themselves, but without the ability to precisely time visual test screenshots I'm not sure how we can really test animations.  I guess we probably can't really, aside from manually.

Reply:
> Hmm I wonder how we can really test the animations. The classes are implementation details, as can be seen by how much they've repeatedly changed between my commits and yours. I'm not inclined to test the classes themselves, but without the ability to precisely time visual test screenshots I'm not sure how we can really test animations. I guess we probably can't really, aside from manually.

yeah... I removed that specific test here: 2dcd76450863e09ba422920e358ac104e35c343e